common/utils/src/main/scala/org/apache/spark/storage/StorageLevel.scala

```diff
@@ -165,21 +165,13 @@ object StorageLevel {
    * Return the StorageLevel object with the specified name.
    */
   @DeveloperApi
-  def fromString(s: String): StorageLevel = s match {
-    case "NONE" => NONE
-    case "DISK_ONLY" => DISK_ONLY
-    case "DISK_ONLY_2" => DISK_ONLY_2
-    case "DISK_ONLY_3" => DISK_ONLY_3
-    case "MEMORY_ONLY" => MEMORY_ONLY
-    case "MEMORY_ONLY_2" => MEMORY_ONLY_2
-    case "MEMORY_ONLY_SER" => MEMORY_ONLY_SER
-    case "MEMORY_ONLY_SER_2" => MEMORY_ONLY_SER_2
-    case "MEMORY_AND_DISK" => MEMORY_AND_DISK
-    case "MEMORY_AND_DISK_2" => MEMORY_AND_DISK_2
-    case "MEMORY_AND_DISK_SER" => MEMORY_AND_DISK_SER
-    case "MEMORY_AND_DISK_SER_2" => MEMORY_AND_DISK_SER_2
-    case "OFF_HEAP" => OFF_HEAP
-    case _ => throw new IllegalArgumentException(s"Invalid StorageLevel: $s")
+  def fromString(s: String): StorageLevel = {
+    try {
+      StorageLevelMapper.fromString(s)
+    } catch {
+      case _: IllegalArgumentException =>
```

Comment:
Just a question. What is the original exception message? I guess there was no clue for the input?

Reply:
The matched `IllegalArgumentException` throws by Java enum, so I convert it to the origin IllegalArgumentException throws by `StorageLevel`'s `fromString`.

Reply:
Yes, I know~ The question was what was the original exception message from Java enum.

Reply:
For example, `StorageLevel.fromString("test")` causes
![image](https://github.com/apache/spark/assets/8486025/3abc7124-a11c-4cfd-88cd-2b51238bfeb0)